ee/modules/650-runtime-audit-engine/templates/k8s-metacollector/deployment.yaml

```diff
@@ -0,0 +1,107 @@
+{{- define "k8s-metacollector_resources" }}
+cpu: 10m
+memory: 25Mi
+{{- end }}
+
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: k8s-metacollector
+  namespace:  d8-{{ $.Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "k8s-metacollector")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: k8s-metacollector
+  updatePolicy:
+    updateMode: "Auto"
+  resourcePolicy:
+    containerPolicies:
+    - containerName: "k8s-metacollector"
+      minAllowed:
+        {{- include "k8s-metacollector_resources" . | nindent 8 }}
+      maxAllowed:
+        cpu: 20m
+        memory: 50Mi
+  {{- end }}
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: k8s-metacollector
+  namespace: d8-{{ $.Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "k8s-metacollector")) | nindent 2 }}
+spec:
+  maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: k8s-metacollector
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: k8s-metacollector
+  namespace: d8-{{ $.Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "k8s-metacollector")) | nindent 2 }}
+spec:
+  {{- include "helm_lib_deployment_strategy_and_replicas_for_ha" . | nindent 2 }}
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: k8s-metacollector
+  template:
+    metadata:
+      labels:
+        app: k8s-metacollector
+    spec:
+      affinity:
+      {{- include "helm_lib_node_selector" (tuple . "system") | nindent 6 }}
+      {{- include "helm_lib_tolerations" (tuple . "system") | nindent 6 }}
+      {{- include "helm_lib_priority_class" (tuple . "cluster-low") | nindent 6 }}
+      {{- include "helm_lib_module_pod_security_context_run_as_user_deckhouse" . | nindent 6 }}
+      serviceAccountName: k8s-metacollector
+      imagePullSecrets:
+      - name: deckhouse-registry
+      containers:
+      - image: {{ include "helm_lib_module_image" (list . "k8sMetacollector") }}
+        name: k8s-metacollector
+        command:
+        - /manager
+        args:
+        - run
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all" . | nindent 8 }}
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8081
+            scheme: HTTP
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+          periodSeconds: 15
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+            scheme: HTTP
+          initialDelaySeconds: 45
+          timeoutSeconds: 5
+          periodSeconds: 15
+        ports:
+        - name: metrics
```

Comment:
we have metrics port. 
Don't we need to protect this port via kube-rbac-proxy?

Reply:
May be, but we do not collect metrics from this pod. Let's discuss.

Reply:
Added kube-rbac-proxy and podMonitor.